cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -112,60 +126,105 @@ func createKubeConfigFiles(outDir string, cfg *kubeadmapi.InitConfiguration, kub
 	return nil
 }
 
-// getKubeConfigSpecs returns all KubeConfigSpecs actualized to the context of the current InitConfiguration
-// NB. this methods holds the information about how kubeadm creates kubeconfig files.
-func getKubeConfigSpecs(cfg *kubeadmapi.InitConfiguration) (map[string]*kubeConfigSpec, error) {
+// createKubeConfigAndCSRFiles creates all the requested kubeconfig files.
+// If kubeconfig files already exists, they are used only if evaluated equal; otherwise an error is returned.
+// This is a variant for use in CreateJoinControlPlaneKubeConfigAndCSRFiles which generates
+// CSR files instead of signed certificates.
+func createKubeConfigAndCSRFiles(kubeConfigDir string, cfg *kubeadmapi.InitConfiguration, kubeConfigFileNames ...string) error {
```

Comment:
i wish we could have avoided some of these refactors to conform unit testing so close to feature freeze...